src/_includes/layouts/base.njk

```diff
@@ -5,6 +5,27 @@
     <title>{% if '/' === page.url %}V8 JavaScript engine{% else %}{{ (renderData.title or title) | markdown | striptags | decodeHtmlEntities }} · V8{% endif %}</title>
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <meta name="color-scheme" content="dark light">
+    <script>
+      // If `prefers-color-scheme` is not supported, fall back to light mode.
+      // In this case, light.css will be downloaded with `highest` priority.
+      if (window.matchMedia('(prefers-color-scheme: dark)').media === 'not all') {
+        document.documentElement.style.display = 'none';
+        document.head.insertAdjacentHTML(
+            'beforeend',
+            '<link rel="stylesheet" href="/_css/light.css" onload="document.documentElement.style.display = \'\'">'
+        );
+      }
+    </script>
+    <!--
+      Conditionally either load the light or the dark stylesheet. The matching file
+      will be downloaded with `highest`, the non-matching file with `lowest`
+      priority. If the browser doesn't support `prefers-color-scheme`, the media
+      query is unknown and the files are downloaded with `lowest` priority (but
+      above I already force `highest` priority for my default light experience).
+    -->
+    <link rel="stylesheet" href="/_css/dark.css" media="(prefers-color-scheme: dark)">
```

Comment:
Given how small these files are and the fact that we download them anyway, is it even worth splitting them up into separate requests? We probably add more overhead than gain anything.

Also, with pure CSS we could use `@supports not (prefers-color-scheme: dark)` as one of the rules for e.g. the light theme so it would nicely fallback without the JS above. Arguably, we could do the same with `@import` inside such condition, but at that point it seems even better to keep it all in one file.

Reply:
> Given how small these files are and the fact that we download them anyway, is it even worth splitting them up into separate requests? We probably add more overhead than gain anything.

+1

> Also, with pure CSS we could use `@supports not (prefers-color-scheme: dark)` as one of the rules for e.g. the light theme so it would nicely fallback without the JS above.

I don't think `@supports` works with media queries. But +1 to doing as much as we can with CSS instead of JS.

Reply:
We've had this discussion before. Just proceed with what you think is the best solution. There're no new arguments to be exchanged. The PR as it was addresses the issue. In one possible way of many.

Reply:
Things [my solution](https://web.dev/prefers-color-scheme/#dark-mode-in-practice) tries to consider:
- Avoiding unused CSS
- Loading files with the right priority
- Making the theme switching easy
- Avoiding flashes of white
- Being backward compatible without causing harm

Reply:
> I don't think @supports works with media queries.

Ah, you mean when combining both in the same rule?

Reply:
`@import` has the huge disadvantage of late discovery. We shouldn't advocate this.

Reply:
> Avoiding unused CSS
> Loading files with the right priority

I understand, but I think metrics not considered here are overhead of individual request (headers etc.), size of individual files after minification (our theme-specific styles are very tiny) as well as compression ratio (it's better when everything is in one file).

In the end, for this particular case it's IMO still benefitial performance-wise to keep things in one file, but I'm happy to be proven wrong by numbers.

Reply:
I’ve been countering request overhead with HTTP/2.

I agree this file architecture might be overkill for v8.dev in the concrete case, but in the general case, people who might take inspiration from v8.dev’s architecture might have waaaay more CSS. 

Again, proceed with what you think is best, my PR just fixed the original issue and makes the toggling work painlessly.

Reply:
> people who might take inspiration from v8.dev’s architecture might have waaaay more CSS

I suppose, but while that's nice, that's not the end goal IMO. We still care about UX for our real users more :)

> Again, proceed with what you think is best, my PR just fixed the original issue and makes the toggling work painlessly.

That's awesome and definitely appreciate that! Just discussing various options to improve before merging.

Reply:
Just to clarify *why* the toggling is painless: the `media` attribute gets changed to activate or deactivate the different stylesheets.

Reply:
Yup, that's clear, thanks! 👍🏻

Reply:
This article just crossed my stream: https://webcloud.se/blog/2020-04-06-flash-of-unstyled-dark-theme/.

Reply:
So could we merge these files back together and proceed like that?

Reply:
So let's take a step back:

Independent from `prefers-color-scheme`, we need to wait until `<dark-mode-toggle>` has [figured out the preferred theme](https://github.com/v8/v8.dev/blob/f9a62400eab6565b2a45eaf1683e38d00bb211c6/src/_js/main.mjs#L54) and potentially applies the `dark` class. Since we render before this happens, the user sees a flash of white. This is avoidable by simply not rendering before the toggle code has run (`root: { display: none; }` and once the toggle has done its work programmatically setting `root: { display: block; }`). Or, and this is what this PR suggests, making the CSS files load conditionally and blockingly with the `media` attribute.

Reply:
I see. I suppose we could instead have dark-mode-toggle add some class to the top element like it's often done (`.loaded` or something) and use that class in CSS condition in addition to `prefers-color-scheme`?

Then we can load CSS like we do today, but it would be applied only when dark-mode-toggle is ready.

Reply:
This would work, too, but you would still need to block the rendering.

Reply:
That's true, even though it unblocks layout at least. Too bad we don't have some sort of stats from the toggle - we could know whether people use it often enough or whether we can render `prefers-color-scheme` variant first and switch later.

Actually, maybe we should do that anyway? As in, while the toggle is not loaded, show variant that depends on `prefers-color-scheme` right away, and switch to the class-based one afterwards?

Then at least first-time visits won't see the flash (because they don't have toggled option yet), and neither will users whose option matches the system-wide theme.

This still leaves users with explicit override, but for them JS should be usually cached already, and the flash would be quick. Overall this is still not ideal but better than current state of things as it reduces or eliminates flash for majority of users?

Reply:
> Actually, maybe we should do that anyway? As in, while the toggle is not loaded, show variant that depends on `prefers-color-scheme` right away, and switch to the class-based one afterwards?

This would not fly with whoever has used the toggle to override their system preference. 

Here's a crazy idea: what if we inline `light.css` and `dark.css` as data URLs (since they're too small to justify an HTTP request), so the `media` attribute-based switching still works?

Reply:
> This would not fly with whoever has used the toggle to override their system preference.

Yes, that's what I said that for them flashing will still occur, but at least everyone else will get non-blocking rendering of the correct theme right away.

> Here's a crazy idea: what if we inline light.css and dark.css as data URLs (since they're too small to justify an HTTP request), so the media attribute-based switching still works?

How is that different from the `.loading` class in terms of behaviour? This seems a bit more complicated to maintain if anything...

Reply:
> How is that different from the `.loading` class in terms of behaviour? This seems a bit more complicated to maintain if anything...

This approach would inherit the native render-blocking behavior of CSS just without the request overhead, and without the need for additional JavaScript.

Reply:
Ah I see what you mean. I think I'd still prefer keeping them together - aside from actual request overhead, splitting provides worse compression and, as mentioned, more complicated maintainability. Let's keep things simple.

Reply:
I close this PR now, since it's definitely not going in the direction that you wanted it to go. It is easier to start from scratch with a "block rendering until toggle has figured out the color scheme" approach.